### ISSUE TITLE
[Feature Request] Generate documentation for a specific view only

### ISSUE THREAD

**Issue by theunsfvo** [archimatetool/archi#294]
This would come in very handy if you have to share details of a specific view with a team, especially in the context of agile solutioning, where views are defined in sprint planning or backlog grooming and you need to share this document of a view with others.

**Comment by MilennialZero** [comment]
Me too. +1. View. Or a folder and everything in a folder. :-)

Jon McLeod
Director
McLeod Consultancy Pty Ltd
GPO Box 813, Canberra ACT 2601 Australia
Silver Member, The Open Group

**Comment by jbsarrodie** [comment]
FWIW you can already generate (jasper based, not HTML) documentation based on tags put on views.

You can find doc on p.85 of the [User Guide](https://www.archimatetool.com/downloads/_release/v4/Archi%20User%20Guide.pdf).

**Comment by theunsfvo** [comment]
I will give it a try, but that seems to be a very round-about way of doing
this. Why not just have a right click / toolbar button to generate it
instantly? Not very use friendly if you need to generate a document per
view or couple of views as we do in Agile Solutioning...


Thanks,
Theuns

"I love deadlines. I like the whooshing sound they make as they fly by."
Douglas Adams
  IQ Business <http://www.iqbusiness.net>: *Solution Architect *| *TOGAF 9
Certified Enterprise Architect *
    My 3D Images <http://bluestarcreations.wordpress.com/> || Some of my
writings <http://www.writingroom.com/writingresults.aspx?m=Zandria> || My
Awesome Stuff Blog <http://awesomestufftheunslikes.wordpress.com/about/>

**Comment by radeksvarz** [comment]
I propose to do that differently. Not view, but selection. If nothing is selected then the whole view.

However there is very fine workaround now, which I started to use.

1. Just select elements on the view. 
2. Copy. Paste to the new model. 
3. Generate documentation from the new model. 

It's fast and you are sure what you are generating documentation for.

**Comment by theunsfvo** [comment]
Had a look at the Jasper way and that only works if you started a model
from a template, according to the documentation.

Radeks way is a lot easier but first prize would be doing it natively
directly from the view you are in or selecting views in the Models pane.

Thanks,
Theuns

"I love deadlines. I like the whooshing sound they make as they fly by."
Douglas Adams
  IQ Business <http://www.iqbusiness.net>: *Solution Architect *| *TOGAF 9
Certified Enterprise Architect *
    My 3D Images <http://bluestarcreations.wordpress.com/> || Some of my
writings <http://www.writingroom.com/writingresults.aspx?m=Zandria> || My
Awesome Stuff Blog <http://awesomestufftheunslikes.wordpress.com/about/>

**Comment by jbsarrodie** [comment]
>Had a look at the Jasper way and that only works if you started a model
from a template, according to the documentation.

No, the model template is just a quick way to pre-set properties on the model to have them already there in the list of properties, but you can create those properties by hand.

>but that seems to be a very round-about way of doing this

I just mentioned it as a workaround. The use-case behind is to be able to create a model containing a big scope (multiple applications, processes...) but being able to generate documentation on a small part of it (per application, per project...).

**Comment by theunsfvo** [comment]
Thanks Jean, but it's way too cumbersome to use. It is quicker to just copy
the diagram and copy the description to word. I'll stick with this method
until something easier and quicker comes along. I know you guys are working
on other feature requests that will probably be higher priority.

Thanks,
Theuns

**Comment by Phillipus** [comment]
> I know you guys are working on other feature requests that will probably be higher priority.

Well, actually we may not necessarily be working on _any_ feature requests at any given time. JB has a full-time job, and I'm doing other things at the moment. The idea is that, as this is an open source project, other people might wish to contribute the code for feature requests.

**Comment by theunsfvo** [comment]
Yeah, I understand how it goes, everybody is busy and work with what they
have available. Would love to get involved but haven't coded in ages and
also have several projects on the go. So I understand how it goes... thanks
Phil.

Thanks,
Theuns

"I love deadlines. I like the whooshing sound they make as they fly by."
Douglas Adams
  IQ Business <http://www.iqbusiness.net>: *Solution Architect *| *TOGAF 9
Certified Enterprise Architect *
    My 3D Images <http://bluestarcreations.wordpress.com/> || Some of my
writings <http://www.writingroom.com/writingresults.aspx?m=Zandria> || My
Awesome Stuff Blog <http://awesomestufftheunslikes.wordpress.com/about/>

**Comment by radeksvarz** [comment]
> other people might wish to contribute the code

Indeed. However that requires skill and time. Which both I am lacking, now. However we as users can complement that with well written feature requests, testing and money.

There are many [Feature Requests] popping up lately. And I have many more in my head.

@Phillipus , have you thought about some Request prioritization / community voting and consequently about some hackaton? Which we as users could support by donating money?

**Comment by Phillipus** [comment]
For this request it's not clear to me whether "generate documentation" refers to the HTML report or the Jasper report. It may be the case that the export wizard could be extended to provide filters for views, folders, elements and so on. In fact, as an example, I believe that someone started a plugin to do something similar.

**Comment by Phillipus** [comment]
> have you thought about some Request prioritization / community voting and consequently about some hackaton? Which we as users could support by donating money?

Yes, JB and I have thought about these types of things. But at the end of the day, somebody has to write the lines of code. And that is usually one person - me.

**Comment by MilennialZero** [comment]
Was going to suggest you start a Patreon page so I could pay a code mercenary five bucks a month to do it for me. :-)

Jon McLeod
Director
McLeod Consultancy Pty Ltd
GPO Box 813, Canberra ACT 2601 Australia
Silver Member, The Open Group

**Comment by jbsarrodie** [comment]
Hi,

> have you thought about some Request prioritization / community voting and consequently about some hackaton? Which we as users could support by donating money?

Well, as you might know I coded several features for Archi, but my main contribution is to (try to) provide vision for Archi. I like to describe myself as a kind of Product Owner (as in Scrum or similar Agile methodologies). This is needed for me as part of my job is to find some people willing to fund new features (that's what happended with Model Repository & Collaboration plugin).

Maybe we could make spread this effort on more people, or at least make it mode visible so at least people could know what Archi could be. I find it very important to have the big picture in mind when added new code in Archi, as we really don't want to waste time and effort.

Then with some clear "feature packs" describe, we could find some ways to fund Phil or other developpers. Patreon or other crowdfunding system being a potential answer.

I think I will explore in the near future solutions (based on GitHub) to group all those small feature requests into manageable "feature pack".

FWIW, These wiki pages could provide a good start:
* [Feature requests and roadmap](https://github.com/archimatetool/archi/wiki/Feature-requests-and-roadmap)
* [Ideas for a nice and elegant way to implement profiles and concepts customization](https://github.com/archimatetool/archi/wiki/Ideas-for-a-nice-and-elegant-way-to-implement-profiles-and-concepts-customization)

**Comment by MilennialZero** [comment]
The more I use Archi (and I've been using it intensively in an applied professional situation for the last two months) - the more I think it would be useful to be able to export a .csv file containing only the elements in a single view. I create what I call "catalogue views" - ie, a view that contains only a graphical list of Archimate business objects. The text description that I capture for each object is of great value. 

I'm using the brute force solution noted by radeksvarz - copying the entities on a diagram into a new model, exporting the .csv, using only the elements.csv file, editing and saving as .xlsx, then importing the elements into a Visio Pro diagram using the Visio "import data to shapes" capability.

We have to use Visio, because customer architects are rarely interested in learning how to use an architecture modelling tool, so they are stuck with static Visio outputs, which have value for a limited period of time.

Even working with a team of professional consultant enterprise architects, very few of them have  competence - or inclination - to use an architecture modelling tool / repository. It's Excel and Visio all the way down.

**Comment by Phillipus** [comment]
I'm working on a plug-in that exports and imports directly to an Excel file. So you could edit the Excel file and re-synch it both ways, in and out.

**Comment by MilennialZero** [comment]
Bi-directional? Want. Happy to pay for it.

**Comment by radeksvarz** [comment]
@MilennialZero I think that export to Visio (or PowerPoint) is such common case that it would benefit the plugin itself.

@Phillipus This is great. I propose to apply the pattern which I used in the properties updater tool ( https://gitlab.com/radek.svarz.dataviso/archi_tools ) 
I.e. to create secondary log sheet when importing from Excel. That saves a lot of headaches.
